src/Controller/UserController.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace BiffBangPow\MessageBoard\Controller;
+
+use BiffBangPow\MessageBoard\FormHandler\UserFormHandler;
+use Doctrine\ORM\EntityRepository;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use BiffBangPow\MessageBoard\Services\SessionService;
+
+class UserController
+{
+    /**
+     * @var \Twig_Environment
+     */
+    private $twig;
+
+    /**
+     * @var EntityRepository
+     */
+    private $userRepository;
+    /**
+     * @var UserFormHandler
+     */
+    private $userFormHandler;
+    /**
+     * @var SessionService
+     */
+    private $sessionService;
+
+    /**
+     * Controller constructor.
+     * @param \Twig_Environment $twig
+     * @param EntityRepository $userRepository
+     * @param UserFormHandler $userFormHandler
+     * @internal param EntityRepository $threadRepository
+     */
+    public function __construct(\Twig_Environment $twig, EntityRepository $userRepository, UserFormHandler $userFormHandler,SessionService $sessionService)
+    {
+        $this->twig = $twig;
+        $this->userRepository = $userRepository;
+        $this->userFormHandler = $userFormHandler;
+        $this->sessionService = $sessionService;
+    }
+
+    /**
+     * @param Request $request
+     * @return Response
+     */
+    public function getLoginAction(Request $request)
+    {
+        return new Response($content = $this->twig->render('login.html.twig', []));
+    }
+
+    /**
+     * @param Request $request
+     * @return RedirectResponse
+     */
+    public function loginAction(Request $request)
+    {
+        $success = $this->userFormHandler->handleLogin($request);
+
+        if ($success) {
+            $this->sessionService->setIsLoggedIn(true);
+
+            return new RedirectResponse('/');
+        } else {
+            throwException($this);
```

Comment:
this is the correct thing to do but in PHP you don't throw exceptions like this. You probably want to do this:
```
throw new \Exception("Login Unsuccessful");
```
You can also create custom exceptions by extending the \Exception class. This means when you catch the exception later in the process you can catch only the exceptions you need to.